src/backend/storage/ipc/ipci.c

```diff
@@ -359,4 +359,29 @@ InitializeShmemGUCs(void)
 		SetConfigOption("shared_memory_size_in_huge_pages", buf,
 						PGC_INTERNAL, PGC_S_DYNAMIC_DEFAULT);
 	}
+
+	/*
+	 * Validating the backend memory limit GUC depends on knowing the size of
+	 * shared memory. This is the first place where we know the size of shmem,
+	 * so it is also the first place we can validate the backend memory limit.
+	 */
+     if (max_total_memory_mb > 0)
+	 {
+		 /* Error if backend memory limit is less than shared memory size */
+		 if (max_total_memory_mb <= size_mb)
+			 ereport(ERROR,
+					 errmsg("configured max_total_memory %dMB is <= shared_memory_size %ldMB",
+							max_total_memory_mb, size_mb),
+					 errhint("Disable or increase the configuration parameter \"max_total_memory\"."));
+
+		 /* Warning if less than 100MB available for non-shared memory */
+		 if (max_total_memory_mb - size_mb < 100)
```

Comment:
Thought we agreed to change this to being reflective of the max_connections setting?

Reply:
Working my way down the list.  I'm not completely happy with the changed version, so it still needs a bit of attention.